platform/Stream.cpp

```diff
@@ -112,14 +103,14 @@ void Stream::rewind() {
 }
 
 int Stream::isatty() {
-    return 0;
+    return 1;
```

Comment:
This change does not break any code. All mbed-os 2 and mbed-os 5 tests are passing, 
I am against touching this again without any solid reason. Use of isatty() has always been internal to us. I don't think we have any external users. If its being deprecated, and everything is working as normal, then there is no need to twiddling things back and forth. We are already a lot behind schedule.